### ISSUE TITLE
[BUG] `Show tags` consider code as a tag

### ISSUE THREAD

**Issue by dmytrodubinin** [renerocksai/telekasten.nvim#115]
**Please confirm**
- I am running the latest version of this plugin

**Describe the bug**
When I try try show tags I get code inside code blocks as a tag, such as `#fff` inside `.css` code block

**To Reproduce**
1. Start nvim
2. run `show tags`
3.  we get all `#...` patterns as a tag even if it is inside code block

**Screenshots**
![image](https://user-images.githubusercontent.com/8066458/165574109-f737a7c2-f5e0-4712-9081-bd63a4be0123.png)
![image](https://user-images.githubusercontent.com/8066458/165575812-7b012a5b-5cd7-4188-9cd0-66e0af74c15c.png)
![Uploading image.png…]()

**Operating system (please complete the following information):**
 - OS: Fedora 35

I think it's worth disabling this behavior by default or giving the opportunity to somehow regulate it

**Comment by lambtho12** [comment]
A quick fix would be to change the [tag notation](https://github.com/renerocksai/telekasten.nvim/discussions/23#discussioncomment-1754511) for telekasten. But I understand that it can be an issues if the notes are to be opened/used with other software that only support `#tag` notation.

I will look how we can check if the tag found is a html color or not and only return the ones that are not html colors.

**Comment by lambtho12** [comment]
Basically the regex should be replaced by something like that. But somehow I can't make it work. It works in the CLI (as long as you add the `--pcre2` flag to `rg`.

`local hashtag_re = "(^|\\s|'|\")((?!(#[a-fA-F0-9]{3})(\W|$)|(#[a-fA-F0-9]{6})(\W|$))#[a-zA-ZÀ-ÿ]+[a-zA-ZÀ-ÿ0-9/\\-_]*)"`

My guess is that the issue is with the escaping of characters in lua, but I do not know what to do from there...

**Comment by lambtho12** [comment]
Of course, I say that I have issues with the regex and 5 minutes later, it finally works!
